odoo_module_migrate/migration_scripts/migrate_130_140.py

```diff
@@ -1,19 +1,153 @@
+    old attribute `src_model=account.move`
+    to `<field name="binding_model_id" ref="account.model_account_move"/>`.
+    """
+    module_name = model_dot_name.split('.')[0]
+    model_und_name = model_dot_name.replace('.', '_')
+    ref_value = f'{module_name}.model_{model_und_name}'
+    field_elem.set('ref', ref_value)
+
+
+def value_to_text(field_elem, attr_value):
+    """Standard behavior: an attribute value is the new fields's text."""
+    field_elem.text = attr_value
+
+
+TAG_ATTR_RENAMING = {
+    'report': {
+        'name': ('report_name', value_to_text),
+        'string': ('name', value_to_text),
+    },
+    'act_window': {
+        'src_model': ('binding_model_id', src_model_new_value),
     }
 }
+"""
+Configuration for renaming particular attributes.
+
+The dictionary maps old tag names to their attributes' names.
+Each old attribute name is then mapped a tuple containing:
+
+- name of the corresponding new field
+- a function that applies the old attribute's value to the new field
+"""
+
+
+def _reformat_file(file_path: Path):
+    """Reformat `file_path`.
+
+    Substitute `act_window` and `report` tag with `record` tag.
+    Note that:
+    - `id` attribute is kept in the `record` tag;
+    - in `report` tag:
+      - `string` has been renamed to `name`;
+      - `name` has been renamed to `report_name`;
+    - other attributes are assigned to respective fields
+    """
+    parser = et.XMLParser(remove_blank_text=True)
+    tree = et.parse(str(file_path.resolve()), parser)
+    root = tree.getroot()
+    reformat_tags = (*root.findall("act_window"), *root.findall("report"))
+    if not reformat_tags:
+        return None
+
+    regexp = r"(?P<indent>[ \t]*)" \
+             r"(?P<tag><{tag_type}[^/]*id=\"{tag_id}\"[^/]*/>)"
+
+    new_tags_dict = dict()
+    for tag in reformat_tags:
+        tag_regex = regexp.format(
+            tag_type=tag.tag,
+            tag_id=tag.attrib['id'],
+        )
+        for attrib, value in tag.attrib.items():
+            if attrib == "id":
+                continue
+            tag.attrib.pop(attrib)
+
+            attr_renames_dict = TAG_ATTR_RENAMING.get(tag.tag, dict())
+            new_value_func = value_to_text
+            if attrib in attr_renames_dict:
+                new_attr_name, new_value_func = attr_renames_dict.get(attrib)
+                attrib = new_attr_name
+
+            field_elem = et.SubElement(tag, "field", {"name": attrib})
+            new_value_func(field_elem, value)
+
+        tag.attrib["model"] = "ir.actions." + tag.tag
+        tag.tag = "record"
+
+        new_tags_dict[tag_regex] = tag
+
+    # Read in the file
+    xml_file = file_path.read_text()
+
+    # Replace the target string
+    for tag_regex, tag in new_tags_dict.items():
+        match = re.search(tag_regex, xml_file)
+        if match:
+            indent = match.group('indent')
+            tag_match = match.group('tag')
+            et.indent(tag, space=indent, level=1)
```

Comment:
You even keep original indent. Perfect :+1:

Reply:
@yelizariev Why do we have to be so careful to only format the modified piece of code, if we then have to pass the "black" tool to format all the files?

https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-14.0#howto
pre-commit run -a  # to run black, isort and prettier (ignore pylint errors at this stage)

Reply:
Because it's easier to check migration changes.
Also, you may want don't run black formating to avoid changes in git history (remember, that this tool can be used outside of OCA)